### DISCUSSION TITLE
QR Code

### DISCUSSION THREAD

**Discussion by kamrankamaei** [QuestPDF/QuestPDF#100] [Q&A] [answered]
What is the best solution to add a QR Code into a document? It seems one way will be to generate the qr code using any other package then add it as an image.

**Comment by MarcinZiabek** [comment, marked as answer]
Hello 😁 There are two possibilities:

1) Generating QR codes as images in a library of your choice. I recommend using a higher resolution to achieve a sharp image. This gives you the highest level of flexibility.

2) Using any SkiaSharp-based library that generates QR codes, [for example this one](https://www.nuget.org/packages/SkiaSharp.QrCode/). To integrate both solutions, you need to use [the Canvas element](https://www.questpdf.com/documentation/api-reference.html#canvas). Please find an example with a similar approach for [generating vector charts](https://www.questpdf.com/documentation/patterns-and-practices.html#implementing-charts).

I am planning to prepare more integration examples in the future. I hope this helps.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response.

Since this discussion was opened, we have added official documentation for integrating QR codes and barcodes into your documents using the ZXing.Net library. You can find detailed examples and usage instructions here:
https://www.questpdf.com/api-reference/barcodes.html

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
